### ISSUE TITLE
Checkbox: Add ariaLabel prop.

### ISSUE THREAD

**Issue by olehnoskov-cengage** [cengage/react-magma#1982]
Add a new optional `ariaLabel` prop to the `Checkbox` component to improve accessibility support.

Original request -> [Link to Slack](https://cengage.slack.com/archives/C8Y4FR63U/p1761310761752479)

**Acceptance Criteria**
- Adopters can pass their own ariaLabel
- If no ariaLabel is passed, we continue to use the existing logic

**Comment by moathabuhamad-cengage** [comment]
Done ✅ 

<img width="1564" height="974" alt="Image" src="https://github.com/user-attachments/assets/0edcbd2b-5f36-45b1-8b4d-3a15191e0df0" />
